library/common/jni/jni_interface.cc

```diff
@@ -1031,3 +1032,127 @@ Java_io_envoyproxy_envoymobile_engine_JniLibrary_setPreferredNetwork(JNIEnv* env
   return set_preferred_network(static_cast<envoy_engine_t>(engine),
                                static_cast<envoy_network_t>(network));
 }
+
+bool jvm_cert_is_issued_by_known_root(JNIEnv* env, jobject result) {
+  jclass jcls_AndroidCertVerifyResult = env->FindClass("org/chromium/net/AndroidCertVerifyResult");
```

Comment:
This call produces a `LocalRef` that must be deleted or it will leak.

Reply:
To clarify, the above is true if the stack originates in the JVM. If the stack originates on a non-JVM-managed thread (e.g. the Envoy Mobile main thread), no allocated references will be automatically cleaned up.

Reply:
Whoa, that should be explicitly stated in the docs (I'm guessing it is and I failed to find it)! Thanks a lot for expanding on this.

For posterity's sake, [Android's JNI tips page](https://developer.android.com/training/articles/perf-jni#local-and-global-references) explicitly mentions this:
> One unusual case deserves separate mention. If you attach a native thread with AttachCurrentThread, the code you are running will never automatically free local references until the thread detaches. Any local references you create will have to be deleted manually.